mocpy/tmoc.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*
+
+"""
+
+tmoc.py:
+  functions to read/write and manipulate TimeMocs
+
+"""
+
+__author__ = "Baumann Matthieu"
+__copyright__ = "CDS, Centre de Données astronomiques de Strasbourg"
+
+try:
+    set
+except NameError:
+    from sets import Set as set
+
+import sys
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import LinearSegmentedColormap
+
+from astropy.time import Time
+
+from .abstract_moc import AbstractMoc
+
+if sys.version > '3':
+    long = int
+
+# Python 3 support
+try:
+    xrange
+except NameError:
+    xrange = range
+
+
+class TimeMoc(AbstractMoc):
+    DAY_MICRO_SEC = 86400000000.
+
+    def __init__(self):
+        AbstractMoc.__init__(self)
+
+    @classmethod
+    def from_table(cls, table, t_column, format='decimalyear'):
+        """
+        Create a tmoc instance from an astropy.table.Table object. The user has to specify the name of the time column
+        to consider.
+
+        :param table: the table containing all the sources we want to build the TimeMoc
+        :param t_column: the name of the column to consider for building the TimeMoc
+        :param format: the format of the times written in this column. See astropy.time.Time
+        for the list of all available formats.
+        :return: the tmoc instance resulting from the sources of the table
+        """
+        if not isinstance(t_column, str):
+            raise TypeError
+
+        moc = TimeMoc()
+
+        times = Time(table[t_column], format=format, scale='tai')
+
+        for time_start in times:
+            moc.add_time(time_start)
+
+        return moc
+
+    @classmethod
+    def from_csv_file(cls, path, **kwargs):
```

Comment:
Astropy `Table.read` supports many CSV and ascii formats. Should save you a little bit of code here and be a bit faster / memory efficient.